[DivideTransformationProfile] generating NumberFormatException

Under certain circumstancies I get this exception at startup : 

````

2022-03-23 14:52:15.684 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
java.lang.NumberFormatException: Invalid BigDecimal value: 0E+1
	at org.openhab.core.library.types.DecimalType.<init>(DecimalType.java:102) ~[?:?]
	at org.openhab.core.library.types.DecimalType.<init>(DecimalType.java:86) ~[?:?]
	at org.openhab.core.library.types.DecimalType.valueOf(DecimalType.java:125) ~[?:?]
	at org.smarthomej.transform.math.internal.profiles.AbstractMathTransformationProfile.transformState(AbstractMathTransformationProfile.java:95) ~[?:?]
	at org.smarthomej.transform.math.internal.profiles.DivideTransformationProfile.onStateUpdateFromHandler(DivideTransformationProfile.java:70) ~[?:?]
	at org.openhab.core.thing.internal.CommunicationManager.lambda$11(CommunicationManager.java:534) ~[?:?]
	at org.openhab.core.thing.internal.CommunicationManager.lambda$13(CommunicationManager.java:554) ~[?:?]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at org.openhab.core.thing.internal.CommunicationManager.handleCallFromHandler(CommunicationManager.java:550) ~[?:?]
	at org.openhab.core.thing.internal.CommunicationManager.stateUpdated(CommunicationManager.java:532) ~[?:?]
	at org.openhab.core.thing.internal.ThingManagerImpl$1.stateUpdated(ThingManagerImpl.java:176) ~[?:?]
	at org.openhab.core.thing.binding.BaseThingHandler.updateState(BaseThingHandler.java:268) ~[?:?]
	at org.openhab.core.thing.binding.BaseThingHandler.updateState(BaseThingHandler.java:287) ~[?:?]
	at org.openhab.binding.synopanalyzer.internal.handler.SynopAnalyzerHandler.lambda$4(SynopAnalyzerHandler.java:161) ~[?:?]
	at java.util.ArrayList.forEach(ArrayList.java:1541) ~[?:?]
	at java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1085) ~[?:?]
	at org.openhab.binding.synopanalyzer.internal.handler.SynopAnalyzerHandler.lambda$3(SynopAnalyzerHandler.java:158) ~[?:?]
	at java.util.Optional.ifPresent(Optional.java:183) ~[?:?]
	at org.openhab.binding.synopanalyzer.internal.handler.SynopAnalyzerHandler.updateSynopChannels(SynopAnalyzerHandler.java:157) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
````

I think it maybe by channel being UNDEF or NULL but not sure.

Correct. The channel value is set to 0 when the exception triggers : 
````
2022-03-29 10:49:00.674 [INFO ] [openhab.event.ItemCommandEvent      ] - Item 'SynopTRAPPES_Octa' received command REFRESH
2022-03-29 10:49:00.742 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'SynopTRAPPES_Octa' changed from 12 to 0

==> /var/log/openhab/openhab.log <==
2022-03-29 10:49:00.735 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.handleCommand()' on 'org.openhab.binding.synopanalyzer.internal.handler.SynopAnalyzerHandler@76c740a5': Invalid BigDecimal value: 0E+1
java.lang.NumberFormatException: Invalid BigDecimal value: 0E+1
	at org.openhab.core.library.types.DecimalType.<init>(DecimalType.java:102) ~[bundleFile:?]
	at org.openhab.core.library.types.DecimalType.<init>(DecimalType.java:86) ~[bundleFile:?]
	at org.openhab.core.library.types.DecimalType.valueOf(DecimalType.java:125) ~[bundleFile:?]
	at org.smarthomej.transform.math.internal.profiles.AbstractMathTransformationProfile.transformState(AbstractMathTransformationProfile.java:95) ~[?:?]
	at org.smarthomej.transform.math.internal.profiles.DivideTransformationProfile.onStateUpdateFromHandler(DivideTransformationProfile.java:70) ~[?:?]
	at org.openhab.core.thing.internal.CommunicationManager.lambda$11(CommunicationManager.java:534) ~[?:?]
	at org.openhab.core.thing.internal.CommunicationManager.lambda$13(CommunicationManager.java:554) ~[?:?]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at org.openhab.core.thing.internal.CommunicationManager.handleCallFromHandler(CommunicationManager.java:550) ~[?:?]
	at org.openhab.core.thing.internal.CommunicationManager.stateUpdated(CommunicationManager.java:532) ~[?:?]
	at org.openhab.core.thing.internal.ThingManagerImpl$1.stateUpdated(ThingManagerImpl.java:176) ~[?:?]
	at org.openhab.core.thing.binding.BaseThingHandler.updateState(BaseThingHandler.java:268) ~[?:?]
	at org.openhab.core.thing.binding.BaseThingHandler.updateState(BaseThingHandler.java:287) ~[?:?]
	at org.openhab.binding.synopanalyzer.internal.handler.SynopAnalyzerHandler.lambda$4(SynopAnalyzerHandler.java:161) ~[?:?]
````

--- a/bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/transform/math/internal/AbstractMathTransformationService.java
+++ b/bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/transform/math/internal/AbstractMathTransformationService.java
@@ -53,8 +53,9 @@ abstract class AbstractMathTransformationService implements TransformationServic
             throw new TransformationException("Math Transformation can only be used with numeric inputs");
         }
         try {
-            String result = performCalculation(source, value).toString();
-            return sourceString.replace(extractedNumericString, result);
+            BigDecimal result = performCalculation(source, value);
+            String resultString = BigDecimal.ZERO.compareTo(result) == 0 ? "0" : result.toString();
+            return sourceString.replace(extractedNumericString, resultString);
         } catch (ArithmeticException e) {
             throw new TransformationException("ArithmeticException: " + e.getMessage());
         }

--- a/bundles/org.smarthomej.transform.math/src/test/java/org/smarthomej/transform/math/internal/DivideTransformationServiceTest.java
+++ b/bundles/org.smarthomej.transform.math/src/test/java/org/smarthomej/transform/math/internal/DivideTransformationServiceTest.java
@@ -57,4 +57,9 @@ class DivideTransformationServiceTest {
     public void testTransformDivideByZero() {
         assertThrows(TransformationException.class, () -> subject.transform("0", "1"));
     }
+
+    @Test
+    public void zeroValueReturnsAParsableValue() throws TransformationException {
+        assertEquals("0", subject.transform("8.0", "0"));
+    }
 }

